### ISSUE TITLE
Extend TitleContainer to enumerate files

### ISSUE THREAD

**Issue by dellis1972** [MonoGame/MonoGame#4542]
I see this allot in games/samples I am porting. The AudioManager of the NetRumble sample is a perfect example. It uses System.IO to enumerate a list of .xnb files which are in a directory. This allows dynamic content of sorts. I know there are other ways (a manifest file for example) but I though this might be a useful addition to the API. At the moment System.IO will NOT work on platforms like android so we end up with platform specific code to grab the "assets" from the Asset manager which is not ideal. when we could easily wrap this up into a useful helper method.

`string[] TitleContainer.GetFiles (string path);`

so we can use something like

`foreach(var sound in TitleContainer.GetFiles("audio") {
      soundsDictionary.Add(sound, Content.Load<SoundEffect>(sound));
}`

Thoughts?

**Comment by KonajuGames** [comment]
TitleContainer generally points to the directory of the executable, and in this we have the one or more content root directories.  So is this to iterate files in potentially one of several content root directories?  This would seem more like a task for the ContentManager to provide a list of files under its own root directory.  Especially in the case you provide where it is using the results of the file list to load assets.  This would only work if a) the Content.RootDirectory was combined with "audio" and b) the correct ContentManager instance was used to load the asset.

**Comment by dellis1972** [comment]
Yes you do need to combine it with the correct ContentManager.RootDirectory I forgot to mention that. So perhaps we need two levels.
The TitleContainer.OpenStream gives low level access to the file system. And the ContentManager uses it to open the files. So perhaps we need both? The low level to get full access and the Higher level per content manager?

**Comment by KonajuGames** [comment]
ContentManager.GetAssets(string path) would be specialized in that it would prefix the path with the RootDirectory, have an implicit "*.xnb" filter and return the names without the `.xnb` extension.  This is so the names returned could be used directly in ContentManager.Load() without further massaging.

TitleContainer.GetFiles(string path) wouldn't have the implicit filter and no massaging of returned names, because sometimes you do want to access non-xnb files in the content folders or their immediate parent.

**Comment by dellis1972** [comment]
Makes sense :) sounds like a plan

**Comment by tomspilman** [comment]
If I remember correctly the reason XNA didn't have something like this to begin was an issue on some platforms... as much as that sounds weird.  Do we need to worry that we might run into platforms where this isn't possible to implement?

Also note that on some platforms these enumerations might return what looks like the same file twice.  This is because of a patch to the game and overlay files.  I suspect on those platforms we want to be sure to filter out the duplicates.

Finally we need to decide what the policy is with case sensitive file systems on this.  I assume the rule is the same as `ContentManager'.. The case needs to match the file/folder correctly on those platforms.
